Formula/pybind11.rb

```diff
@@ -34,10 +34,11 @@ def install
 
     pythons.each do |python|
       # Install Python package too
-      system python.opt_bin/"python3", *Language::Python.setup_install_args(libexec)
+      site_packages = Language::Python.site_packages python.opt_bin/"python3"
+      system python.opt_bin/"python3", *Language::Python.setup_install_args(libexec),
+                                       "--install-lib=#{libexec/site_packages}"
```

Comment:
Tagged `do not merge` so that someone can check if possible to fix in `python@3.10` formula or if we need to manually do workarounds for all older `distutils`-based installations.